### PR TITLE
Support eos_token_id being an array

### PR DESCRIPTION
**Pull request by RyanUnderhill**
Some models have the eos_token_id being an array of tokens vs a single value. This adds support for it by processing the logits in such a way that the first eos_token_id in the array gets the max score of all of the ones in the array.

This way, if any of the eos_token_id scores is high scoring, the primary eos_token_id score will get that score and the scoring code will check for it as before.

### CODE REVIEW

File: src/models/model.cpp [hunk 1]
```diff
@@ -537,6 +537,7 @@ std::unique_ptr<OrtValue> Model::ExpandInputs(std::unique_ptr<OrtValue>& input,
 }
 
 void Model::GetMaxBatchSizeFromGeneratorParams(const GeneratorParams& params) {
```
Comment (yufenglee) [inline src/models/model.cpp:539]: >GetMaxBatchSizeFromGeneratorParams [](http://example.com/codeflow?start=12&length=34)

why is it commented out?

File: src/models/logits.cpp [hunk 2]
```diff
@@ -126,33 +139,58 @@ RoamingArray<float> Logits::Get() {
...
+  size_t vocab_index = 0;  // Simpler math to have this index go up by vocab_size for every logit chunk we process
+
+  for (int index = 0; index < shape_[0]; index++) {
+    auto logits = batched_logits.subspan(vocab_index, vocab_size);
+    float max = std::numeric_limits<float>::lowest();
+    for (auto id : model_.config_->model.eos_token_ids)
+      max = std::max(max, logits[id]);
+    logits[model_.config_->model.eos_token_id] = max;  // Set the score of the primary EOS token to the highest of any of the EOS tokens
+    vocab_index += vocab_size;
+  }
```
Comment (yufenglee) [inline src/models/logits.cpp:193]: does it work for sampling?

File: src/models/model.cpp (hunk 1 above)
Comment (RyanUnderhill) [reply src/models/model.cpp]: Whoops, that wasn't supposed to be merged.

File: src/models/logits.cpp (hunk 2 above)
Comment (RyanUnderhill) [reply src/models/logits.cpp:196]: It does now

**Review by yufenglee — APPROVED** [review]
:shipit:

**Review by kunal-vaishnavi — APPROVED** [review]
